newIDE/app/src/MainFrame/index.js

```diff
@@ -1742,8 +1746,14 @@ class MainFrame extends React.Component<Props, State> {
                 this.renameEventsFunctionsExtension
               }
               onRenameExternalEvents={this.renameExternalEvents}
-              onSaveProject={this.saveProject}
-              onSaveProjectAs={this.saveProjectAs}
+              onSaveProject={() => {
+                this.context.sealUnsavedChanges();
```

Comment:
Should we do this here or inside this.saveProject/this.saveProjectAs?

I'm worried about other code calling (now or in the future) this.saveProject and so not sealing the unsaved changes.

Reply:
I used it here too because when you use the browser editor and you download the file, the message "Project properly saved" (saving flag) doesn't show up so the context doesn't update. I will check a little more the data flow between MainFrame and the saving Operators.